spec/lcm/integration/user_provisioning_end_to_end_spec.rb

```diff
@@ -0,0 +1,215 @@
+      Support::S3Helper.upload_file(users_csv, @test_context[:users_brick_input][:s3_key])
+
+      atts = @projects.first.attributes.to_a
+      $attribute = atts.shift while !$attribute || $attribute.date_attribute? || !$attribute.primary_label
+      label = $attribute.primary_label
+      muf_template = {}
+      element = label.values.find { |v| !v[:value].empty? }
+      muf_template[label.title] = element[:value]
+      $mufs = $user_data.map do |u|
+        m = muf_template.dup
+        m[:login] = u[:custom_login]
+        m[:project_id] = u[:project_id]
+        m
+      end.flatten
+
+      upload_user_filters_csv($mufs)
+    end
+
+    it 'setups the MUFs' do
+      config_path = ConfigurationHelper.create_interpolated_tempfile(
+        @ufb_template_path,
+        @test_context
+      )
+
+      $SCRIPT_PARAMS = JSON.parse(File.read(config_path))
+      GoodData::Bricks::Pipeline.user_filters_brick_pipeline.call($SCRIPT_PARAMS)
+      @projects.each do |p|
+        filters = p.user_filters.to_a
+        expect(filters.length).to eq $user_amount
+        @users.each do |u|
+          expect(filters.find { |f| f.related.email == u })
+        end
+      end
+    end
+  end
+
+  describe '2 - initial user setup' do
+    it 'setups the users' do
+      config_path = ConfigurationHelper.create_interpolated_tempfile(
+        @ub_template_path,
+        @test_context
+      )
+
+      $SCRIPT_PARAMS = JSON.parse(File.read(config_path))
+      GoodData::Bricks::Pipeline.users_brick_pipeline.call($SCRIPT_PARAMS)
+      @projects.each do |p|
+        users = p.users.to_a
+        expect(users.length).to eq ($user_amount + 1) # rubydev+admin is in the projects as well (as the admin)
+      end
+    end
+  end
+
+  describe 'modify the user input' do
+    it 'does it' do
+      user_data = $user_data.map do |c|
+        cdup = c.dup
+        cdup['position'] = 'manager'
+        cdup
+      end.flatten
+      users_csv = ConfigurationHelper.csv_from_hashes(user_data)
+      Support::S3Helper.upload_file(users_csv, @test_context[:users_brick_input][:s3_key])
+    end
+  end
+
+  describe 'modify the MUF input' do
+    it 'does it' do
+      atts = @projects.first.attributes.to_a
+      attribute = atts.shift while !attribute || attribute.date_attribute? || !attribute.primary_label || attribute == $attribute
+      label = attribute.primary_label
+      element = label.values.find { |v| !v[:value].empty? }
+
+      mufs = $mufs.map do |m|
+        mdup = m.dup
+        mdup[label.title] = element[:value]
+        mdup
+      end
+
+      upload_user_filters_csv(mufs)
+    end
+  end
+
+  describe '3 - subsequent MUF setup' do
+    it 'checks if they appeared' do
+      @projects.each do |p|
+        filters = p.user_filters.to_a
+        expect(filters.length).to eq $user_amount
+        @users.each do |u|
+          expect(filters.find { |f| f.related.email == u })
+        end
+      end
+    end
+
+    it 'pushes them there' do
```

Comment:
Similar code found in user_provisioning_end_to_end_spec.rb:116, user_provisioning_end_to_end_spec.rb:190 (mass = 100)